voter/management/commands/voter_process_snapshot.py

```diff
@@ -126,94 +151,135 @@ def reset_file(file_tracker):
     file_tracker.save()
 
 
+def flush():
+    """Bulk insert pending NCVoter and ChangeTracker rows. Also, clear all such
+    buffer lists.
+    """
+    with transaction.atomic():
+        if voter_records:
+            NCVoter.objects.bulk_create(voter_records)
+    with transaction.atomic():
+        # This looks weird. Let me explain.
+        # All the unsaved ChangeTracker instances have references
+        # to the NCVoter instances from *before* the NCVoter instances
+        # were saved. So they do not know the voter instances now have
+        # IDs from being inserted. This re-sets the voter on the change
+        # object, ensuring it knows the ID of its voter and can be saved
+        # properly.
+        for c in change_records:
+            c.voter = c.voter
+            c.voter_id = c.voter.id
+        ChangeTracker.objects.bulk_create(change_records)
+    reset()
+
+
+def reset():
+    change_records.clear()
+    voter_records.clear()
+    processed_ncids.clear()
+
+
+def skip_or_voter(row):
+    global skip_tally
+    global ignored_tally
+
+    # If we see a repeat, flushed queued data before continuing
+    # This prevents the same voter from appearing twice in a single bulkd insert
+    ncid = row.get('ncid')
+    if ncid in processed_ncids:
+        flush()
+    voter_instance = find_existing_instance(ncid)
+
+    # Skip rows that have no NCID in them :-(
+    # TODO: Log these so we can come back and figure out what to do with them
+    if ncid is None:
```

Comment:
Would it be appropriate to change this to `if not ncid`? We'd want to raise the error for empty strings as well.